sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/DoNotHardcodeCredentialsBase.cs

```diff
@@ -53,8 +53,10 @@ public string CredentialWords
                 this.splitCredentialWords = value.ToUpperInvariant()
                     .Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries)
                     .Select(x => x.Trim())
+                    .Select(Regex.Escape)
```

Comment:
`FindCredentialWords` is using `splitCredentialWords` without regex to generate `credentialWordsFound`. We should move the `.Select(Regex.Escape)` to the next statement with RegEx creation for consistent behavior.

Reply:
This is quite a corner case since the escaped characters cannot be part of the identifier name. I would let this as it is since it's a little bit easier to read.

Reply:
It is a corner case, but I don't like it for general principle. The variable `splitCredentialWords` should contain words, otherwise it has bad name and that's dangerous. Moving .Select() to statement below is not difficult to read and it's clear sanitizer for the regex.

Reply:
I've moved it to the other line but the characters we are escaping are not words and will not remain words if we are not escaping them.